eng/pipelines/templates/jobs/ci.yml

```diff
@@ -351,6 +351,27 @@ jobs:
           publishJUnitResults: false
           goals: 'verify'
 
+      - script: |
+          npm install -g autorest
+        displayName: 'Prepare Environment for Generation'
+
+      - task: PowerShell@2
+        displayName: Verify Code Generation
+        inputs:
+          pwsh: true
+          filePath: $(Build.SourcesDirectory)/eng/pipelines/scripts/Invoke-Codegeneration.ps1
+          arguments: >
+            -Directory sdk/${{ parameters.ServiceDirectory }}
+
+      - script: |
```

Comment:
I would suggest adding this git diff and error as part of the script. See what .NET does for example https://github.com/Azure/azure-sdk-for-net/blob/main/eng/scripts/CodeChecks.ps1#L171

Reply:
Thanks for that link, I was just copying what was being done in the autorest.java repo and that looks like a better choice.